tests/test_scalars.py

```diff
@@ -708,6 +805,293 @@ def test_timestamp() -> None:
+            pd.Series([1, 2, 3], dtype="datetime64[ns]") <= ts, "pd.Series[bool]"
+        ),
+        pd.Series,
+    )
+
+    check(assert_type(ts == ts, bool), bool)
+    check(assert_type(ts == np.datetime64(1, "ns"), bool), bool)
+    check(assert_type(ts == dt.datetime(year=2000, month=1, day=1), bool), bool)
+    check(
+        assert_type(ts == pd.DatetimeIndex(["2000-1-1"]), np_ndarray_bool), np.ndarray
+    )
+    check(
+        assert_type(ts == np_dt64_arr, np_ndarray_bool),
+        np.ndarray,
+    )
+    check(
+        assert_type(
+            ts == pd.Series([1, 2, 3], dtype="datetime64[ns]"), "pd.Series[bool]"
+        ),
+        pd.Series,
+    )
+
+    check(assert_type(dt.datetime(year=2000, month=1, day=1) == ts, bool), bool)
+    check(
+        assert_type(pd.DatetimeIndex(["2000-1-1"]) == ts, np_ndarray_bool), np.ndarray
+    )
+
+    check(
+        assert_type(
+            pd.Series([1, 2, 3], dtype="datetime64[ns]") == ts, "pd.Series[bool]"
+        ),
+        pd.Series,
+    )
+
+    check(assert_type(ts != ts, bool), bool)
+    check(assert_type(ts != np.datetime64(1, "ns"), bool), bool)
+    check(assert_type(ts != dt.datetime(year=2000, month=1, day=1), bool), bool)
+    check(
+        assert_type(ts != pd.DatetimeIndex(["2000-1-1"]), np_ndarray_bool), np.ndarray
+    )
+    check(assert_type(ts != np_dt64_arr, np_ndarray_bool), np.ndarray)
+    check(
+        assert_type(
+            ts != pd.Series([1, 2, 3], dtype="datetime64[ns]"), "pd.Series[bool]"
+        ),
+        pd.Series,
+    )
+
+    check(assert_type(dt.datetime(year=2000, month=1, day=1) != ts, bool), bool)
+    check(
+        assert_type(pd.DatetimeIndex(["2000-1-1"]) != ts, np_ndarray_bool), np.ndarray
+    )
+    check(
+        assert_type(
+            pd.Series([1, 2, 3], dtype="datetime64[ns]") != ts, "pd.Series[bool]"
+        ),
+        pd.Series,
+    )
+
+    # Failures due to NumPy ops returning Any
+    check(
+        assert_type(  # type: ignore[assert-type]
+            np_td64_arr + ts, npt.NDArray[np.datetime64]  # type: ignore[operator]
+        ),
+        np.ndarray,
+    )
+    check(assert_type(np.datetime64(1, "ns") > ts, np.bool_), np.bool_)  # type: ignore[assert-type]
+    check(assert_type(np.datetime64(1, "ns") >= ts, np.bool_), np.bool_)  # type: ignore[assert-type]
+    check(assert_type(np.datetime64(1, "ns") < ts, np.bool_), np.bool_)  # type: ignore[assert-type]
+    check(assert_type(np.datetime64(1, "ns") <= ts, np.bool_), np.bool_)  # type: ignore[assert-type]
+    check(assert_type(np.datetime64(1, "ns") == ts, np.bool_), np.bool_)  # type: ignore[assert-type]
+    check(assert_type(np_dt64_arr == ts, np_ndarray_bool), np.ndarray)  # type: ignore[assert-type]
+    check(assert_type(np.datetime64(1, "ns") != ts, np.bool_), np.bool_)  # type: ignore[assert-type]
+    check(assert_type(np_dt64_arr != ts, np_ndarray_bool), np.ndarray)  # type: ignore[assert-type]
+
+
+def test_types_init() -> None:
```

Comment:
I know you copied these over from `test_timefuncs.py`, but since they are now mixed with the other scalars, can you rename the functions to have `timestamp` in it?

Reply:
These have been renamed in one of the PRs.